milli/src/search/new/query_term/parse_query.rs

```diff
@@ -79,10 +79,10 @@ pub fn located_query_terms_from_tokens(
             TokenKind::Separator(separator_kind) => {
                 match separator_kind {
                     SeparatorKind::Hard => {
-                        position += 1;
+                        position = position.wrapping_add(1);
                     }
                     SeparatorKind::Soft => {
-                        position += 0;
+                        position = position.wrapping_add(0); // TODO: this is a no-op. Consider removing?
```

Comment:
don't you want to do it in this PR? By replacing it with an `if let ...`?

Reply:
Changed to a `if let ...`